MediaBrowser.Common/Net/NetworkExtensions.cs

```diff
@@ -9,240 +13,351 @@ namespace MediaBrowser.Common.Net
+        public static string FormatIpString(IPAddress? address)
         {
-            if (source.Count == 0)
+            if (address is null)
             {
-                return false;
+                return string.Empty;
             }
 
-            ArgumentNullException.ThrowIfNull(item);
-
-            foreach (var i in source)
+            var str = address.ToString();
+            if (address.AddressFamily == AddressFamily.InterNetworkV6)
             {
-                if (i.Contains(item))
+                int i = str.IndexOf('%', StringComparison.Ordinal);
+                if (i != -1)
                 {
-                    return true;
+                    str = str.Substring(0, i);
                 }
+
+                return $"[{str}]";
             }
 
-            return false;
+            return str;
         }
 
         /// <summary>
-        /// Compares two Collection{IPObject} objects. The order is ignored.
+        /// Try parsing an array of strings into <see cref="IPNetwork"/> objects, respecting exclusions.
+        /// Elements without a subnet mask will be represented as <see cref="IPNetwork"/> with a single IP.
         /// </summary>
-        /// <param name="source">The <see cref="Collection{IPObject}"/>.</param>
-        /// <param name="dest">Item to compare to.</param>
-        /// <returns>True if both are equal.</returns>
-        public static bool Compare(this Collection<IPObject> source, Collection<IPObject> dest)
+        /// <param name="values">Input string array to be parsed.</param>
+        /// <param name="result">Collection of <see cref="IPNetwork"/>.</param>
+        /// <param name="negated">Boolean signaling if negated or not negated values should be parsed.</param>
+        /// <returns><c>True</c> if parsing was successful.</returns>
+        public static bool TryParseToSubnets(string[] values, out List<IPNetwork> result, bool negated = false)
         {
-            if (dest is null || source.Count != dest.Count)
+            result = new List<IPNetwork>();
+
+            if (values is null || values.Length == 0)
             {
                 return false;
             }
 
-            foreach (var sourceItem in source)
+            for (int a = 0; a < values.Length; a++)
             {
-                bool found = false;
-                foreach (var destItem in dest)
+                string[] v = values[a].Trim().Split("/");
+
+                var address = IPAddress.None;
+                if (negated && v[0].StartsWith('!'))
                 {
-                    if (sourceItem.Equals(destItem))
-                    {
-                        found = true;
-                        break;
-                    }
+                    _ = IPAddress.TryParse(v[0][1..], out address);
+                }
+                else if (!negated)
+                {
+                    _ = IPAddress.TryParse(v[0][0..], out address);
```

Comment:
You should check the result instead of discarding it. You can store `v[0][0..]`/`v[0][1..]` in a variable to avoid calling the method in two separate places